version: "3.5"

services:
  plex:
    container_name: plex
    image: plexinc/pms-docker:1.42.1.10060-4e8b05daf
    hostname: ${APP_DOMAIN}
    restart: on-failure:5
    environment:
      # General
      - TZ=Europe/Paris
      # NGINX + LETSENCRYPT
      - VIRTUAL_PORT=32400
      - VIRTUAL_HOST=${APP_DOMAIN}
      - LETSENCRYPT_HOST=${APP_DOMAIN}
      # APP
      - PLEX_CLAIM=${PLEX_CLAIM}
    ports:
      - "32400:32400/tcp"
    networks:
      - proxy
      - plex
    devices:
      - "/dev/dri:/dev/dri"
    volumes:
      - ./volumes/config:/config # plex database
      - ./volumes/transcode:/transcode # plex transcode
      - ~/files/movies:/data/movies
      - ~/files/series:/data/series
      - ~/files/music:/data/music
      - ~/files/videos:/data/videos
      - ~/files/photos:/data/photos

  # threadfin:
  #   hostname: threadfin
  #   image: fyb3roptik/threadfin:1.2.32
  #   restart: on-failure:5
  #   ports:
  #     - 127.0.0.1:34400:34400
  #   environment:
  #     # General
  #     - TZ=Europe/Paris
  #   volumes:
  #     - ./volumes/threadfin:/home/threadfin/conf
  #   networks:
  #     - plex

networks:
  proxy:
    external: true
  plex:
